compiler/src/dotty/tools/dotc/transform/Erasure.scala

```diff
@@ -58,6 +58,18 @@ class Erasure extends Phase with DenotTransformer {
           }
         }
 
+      def erasedName =
+        if ref.is(Flags.Method)
+            && contextResultsAreErased(ref.symbol)
+            && (ref.owner.is(Flags.Trait) || ref.symbol.allOverriddenSymbols.hasNext)
```

Comment:
Name mangling is always bad because it confuses tack traces and (in this case) causes more indirection via bridge methods, which blows up code sizes. So I'd avoid it if we don't absolutely need it.